common/src/main/java/whocraft/tardis_refined/common/tardis/control/flight/CoordinateControl.java

```diff
@@ -44,15 +48,16 @@ private boolean changeCoord(TardisLevelOperator operator, ConsoleTheme theme, Co
 
             int increment = pilotManager.getCordIncrement();
             int incrementAmount = addValue ? increment : -increment;
-            BlockPos potentialPos = pilotManager.getTargetLocation().getPosition();
+            TardisNavLocation targetLocation = pilotManager.getTargetLocation();
+            BlockPos potentialPos = targetLocation.getPosition();
 
             switch (button){
                 case X -> potentialPos = potentialPos.offset(incrementAmount, 0, 0);
                 case Y -> potentialPos = potentialPos.offset(0, incrementAmount, 0);
                 case Z -> potentialPos = potentialPos.offset(0, 0, incrementAmount);
             }
 
-            if (pilotManager.getTargetLocation().getLevel().isInWorldBounds(potentialPos)){ //Use vanilla check which accounts for both world height and horizontal bounds
+            if ((targetLocation.getLevel().dimension() == Level.END || targetLocation.getLevel().isInWorldBounds(potentialPos)) && !pilotManager.getTargetLocation().getLevel().isOutsideBuildHeight(potentialPos)){ //Use vanilla check which accounts for both world height and horizontal bounds
```

Comment:
Needs some documentation for better clarity.

1. For the first part of the logic, are we trying to say that we are allowing players to change coordinates when the target location is in the End dimension?

2. Regarding the first part of the logic, do we need the ``isInWorldBounds`` regardless of whether the target location is the End dimension?
`(
targetLocation.getLevel().dimension() == Level.END || targetLocation.getLevel().isInWorldBounds(potentialPos)
)`
suggests that if the target location is in the End (first condition = TRUE), then the ``isInWorldBounds`` check won't be used. 

3. For second part of logic, shall we we consider using ``isInWorldBounds`` instead of ``isOutsideBuildHeight``? ``isInWorldBounds`` already calls ``isOutsideBuildHeight`` but also runs an additional check for valid horizontal positions. Hence ``isInWorldBounds`` would be a more comprehensive check, unless we are intentionally only considering a vertical height check, in which case ``isOutsideBuildHeight`` would be enough.

Reply:
There was a bug here that was designed to resolve an issue regarding the end. I can no longer recreate the bug. Ignore this logic as I am reverting it.